docs/advanced_features.rst

```diff
@@ -13,74 +13,109 @@ This guide will cover the following advanced use cases:
 
-Then, when defining our source for this type of event we would include
-a path to this function in the ``context_loader`` field.
+In order to render the event with this template, we first set up a rendering style. This rendering
+style is pretty short and could probably be displayed in many places that want to display short
+messages (like a notification bar). So, we can make a ``short`` rendering style as followings:
 
 .. code-block:: python
 
-    from entity_event import Source
+    short_rendering_style = RenderingStyle.objects.create(
+        name='short',
+        display_name='Short Rendering Style')
 
-    photo_tag_source = Source.objects.create(
-        name="photo-tag",
-        display_name="Photo Tag",
-        description="You are tagged in a photo",
-        group=photo_group,
-        context_loader='apps.photos.loaders.load_photo_path'
+Now that we have our rendering style, we need to create a context renderer that has information about
+what templates, source, rendering style, and context hints to use when rendering the event. In our case,
+it would look like the following:
+
+.. code-block:: python
+
+    context_renderer = ContextRenderer.objects.create(
+        render_style=RenderingStyle.objects.get(name='short'),
+        name='short_login_renderer',
+        html_template_path='my_template_dir/user_logged_in.html',
+        source=Source.objects.get(name='user_logged_in'),
+        context_hints={
+            'user': {
+                'app_name': 'auth',
+                'model_name': 'User',
+            }
+        }
     )
 
-With this setup, all of the additional information can by dynamically
-loaded into events, simply by calling :py:meth:`Event.get_context
-<entity_event.models.Event.get_context>`.
-
-The ``Source`` model also uses django's ``clean`` method to ensure
-that only valid importable functions get saved in the
-database. However, if this function is removed from the codebase,
-without a proper migration, attempting to load context for events with
-this source will fail.
-
-There are a number of trade-offs in using a context loader. If the
-underlying data is subject to change, accessing historic events could
-cause errors in the application. Additionally, a context loader that
-requires many runs to the database could cause accessing events to be
-a much more expensive operation. In either of these cases it makes
-more sense to store copies of the data in the ``context`` field of the
-event.
+In the above, we set up the context renderer to use the short rendering style, pointed it to our html template
+that we created, and also pointed it to the source of the event. As you can see from the html template, we
+want to reach inside of the Django User object and display the ``username`` field. In order to retrieve this
+information, we have told our context renderer to treat the ``user`` key from the event context as a PK
+to a Django ``User`` model that resides in the ``auth`` app.
```

Comment:
This is a great example, and is extremely helpful in understanding the workflow. I think it would be a good idea to note that the full documentation for what sorts of hints can be created is available in the documentation for `models.ContextRenderer`. A link to those docs would probably be a good idea as well.